library/lib.sh

```diff
@@ -477,62 +508,69 @@ lsrDisableNFV() {
     fi
 }
 
-lsrGenerateTestDisks() {
-# This function generates test disks from provision.fmf
-# This is required by storage and snapshot roles
-    local provisionfmf
-    local -i i=0
-    local disk_provisioner_dir TARGETCLI_CMD disks disk file
-    lsrGetRoleDir
-    provisionfmf="$role_path"/tests/provision.fmf
-    if [ ! -f "$provisionfmf" ]; then
-        rlRun "rm -rf ${role_path}"
-        return
+lsrDiskProvisionerRequired() {
+# check if the test requires additional disk devices
+# to be provisioned
+    local tests_path=$1
+    local provision_fmf="$tests_path/provision.fmf"
+    if grep -q "drive:" "$provision_fmf" 2> /dev/null; then
+        return 0
     fi
-    if ! grep -q drive: "$provisionfmf"; then
-        rlRun "rm -rf ${role_path}"
-        return
+    return 1
+}
+
+lsrCheckPartitionSize() {
+    local file_or_dir="$1"
+    local op="$2"
+    local size="$3"  # in kb - like df -k
+    local available
+    available=$(df -k "$file_or_dir" --output=avail | tail -1)
+    test "$available" "$op" "$size"
+}
+
+lsrGenerateTestDisks() {
+    local tests_path=$1
+    local provisionfmf="$tests_path"/provision.fmf
+    local disk_provisioner_script=disk_provisioner.sh
+    local disk_provisioner_dir
+    if ! lsrDiskProvisionerRequired "$tests_path"; then
+        return 0
     fi
-    rlRun "yum install targetcli -y"
-    disks=$(sed -rn 's/^\s*-?\s+size:\s+(.*)/\1/p' "$provisionfmf")
-    # Nothing to do
-    [ -z "$disks" ] && return
 
     # disk_provisioner needs at least 10GB - if /tmp does not have enough space, use /var/tmp
     if lsrCheckPartitionSize "/tmp" -gt 10485760; then
-        disk_provisioner_dir=$(mktemp --directory --tmpdir=/tmp)
+        disk_provisioner_dir=/tmp/disk_provisioner
     else
-        disk_provisioner_dir=$(mktemp --directory --tmpdir=/var/tmp)
+        disk_provisioner_dir=/var/tmp/disk_provisioner
```

Comment:
The disk is provisioned on the managed node?  Does lsrCheckPartitionSize check the managed node?